### DISCUSSION TITLE
Adding BitBadges to bitclout.com frontend

### DISCUSSION THREAD

**Discussion by trevormil** [deso-protocol/dips#105] [Medium CIPs]
**What is BitBadges?**  

BitBadges is an open source, community driven platform where BitClout users can issue NFT badges to other users!

Like BitClout laid the foundation by providing public access to data like posts, messages, transactions, etc, BitBadges lays the foundation by providing public access to badges. We offer a public, open source API with access to all data for all badges so anyone can build whatever they wish on top of our ecosystem.

**What would we like to add?** 

We would like to add support for issuing badges and displaying one's earned badges on their profile to the bitclout.com frontend. We have already launched on CloutFeed and would like to add all the same functionality here for desktop users.

Note: @trevormil will happily do all the work and create a PR for this, but we are looking to get approval first before I waste any time.

**What are badges?**

Badges take inspiration from NFT tokens, but the main difference between them is that badges are tied to a public key and non transferable. So once you earn one, no one can ever take it away from you!

**What is our long term vision and how will this help BitClout?**

In the short term, we offer a fun and unique feature to BitClout for users to engage and interact with their community. Badges can be issued for anything from loyal fan badges to access tickets to verification checks.

In the long term, we aim to be the one stop shot for cross chain NFTs. For example, we are working on implementing a way to issue a badge NFT from your $CLOUT wallet to someone's $ETH wallet. Eventually, the vision is that we want everyone to have the ability to link their wallets to their $CLOUT profile, and then, your $CLOUT profile becomes your digital identity for all wallets. Just like you go onto OpenSea and see all your ERC-721 NFTs for your wallet, you would be able to go onto BitClout and all in one place, see your $CLOUT NFTs, ERC-721 NFTs, badges issued to your $CLOUT Profile, and badges issued to your $ETH profile.

**How is this implemented?**

Currently, we have our own open source API where we use BitClout Identity to verify users are who they say they are. All badge data is stored on IPFS, and the file hash of that data is posted on the BitClout chain via a posting account @BitBadgesHash. So, all badge data is stored in three places (IPFS, database, and @BitBadgesHash) to always verify that we are being honest and can always be trusted. We would also love to discuss about storing this data on the BitClout chain legitly as well.

Documentation: bitbadges.github.io
Source Code: https://github.com/BitBadges

Happy to answer any questions! Let me know what you think!

**Comment by danwils33** [comment]
game changer! let's get it done!

**Comment by tijno** [comment]
I think there needs to be consideration given to what 3rd party apis are integrated, and how.

And with bitclouts stated intention to go away in future - would it make more sense to make this super easy for node operators to add to their current nodes?

Eg if it was an easy change - that doesnt make it hard to merge with the main upstream branch changes daily - i would add it to tijn.club for sure.
